### PR TITLE
operator: log the cancel reason

### LINKED ISSUE #6677: operator: log the cancel reason
## Enhancement Task
In current, operator must pass some checker before it runs. It maybe cancel because one of  the checkers is failed, it's too hard to know the reason.
```
[2023/06/09 12:20:06.610 +08:00] [INFO] [operator_controller.go:611] ["operator canceled"] [region-id=1293] [takes=0s] [operator="\"scatter-region {mv light peer: store [10] to [4]} (kind:admin,region,leader, region:1293(191, 25), createAt:2023-06-09 12:20:06.610477352 +0
800 CST m=+1961.716996864, startAt:0001-01-01 00:00:00 +0000 UTC, currentStep:0, size:22, steps:[add learner peer 1299 on store 4, use joint consensus, promote learner peer 1299 on store 4 to voter, demote voter peer 1294 on store 10 to learner, transfer leader from store
 10 to store 4, leave joint state, promote learner peer 1299 on store 4 to voter, demote voter peer 1294 on store 10 to learner, remove peer on store 10]，timeout:[18m0s])\""] [additional-info=]
```

### PR DESCRIPTION
**Pull request by bufferflies**
### What problem does this PR solve?

Issue Number: Close #6677 , #6605

### What is changed and how does it work?

```commit-message
log the cancel reason
```

### Check List

Tests

- Unit test
- Integration test
- Manual test

```
 [2023/06/26 16:30:54.524 +08:00] [INFO] [operator_controller.go:641] ["operator canceled"] [region-id=1] [takes=8.001042ms] [operator="\"admin-add-peer {add peer: store [3]} (kind:admin,region, region:1(1, 1), createAt:2023-06-26 16:30:54.516444 +0800 CST m=+3.298618835, startAt:2023-06-26 16:30:54.516797 +0800 CST m=+3.298972543, currentStep:0, size:1, steps:[add learner peer 1 on store 3, promote learner peer 1 on store 3 to voter]，timeout:[11m0s])\""] [additional-info="{\"cancel-reason\":\"admin stop\"}"]
```
Code changes

Side effects

Related changes

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/schedule/operator/operator_controller.go [hunk 1]
```diff
@@ -609,7 +637,7 @@ func (oc *Controller) buryOperator(op *Operator, extraFields ...zap.Field) {
 			zap.Reflect("operator", op),
 			zap.String("additional-info", op.GetAdditionalInfo()),
 		}
-		fields = append(fields, extraFields...)
+		fields = append(fields)
```
Comment (HuSharp) [inline pkg/schedule/operator/operator_controller.go:640]:
```suggestion
```

File: pkg/schedule/operator/operator_controller.go [hunk 2]
```diff
@@ -327,9 +330,17 @@ func (oc *Controller) AddOperator(ops ...*Operator) bool {
 	// note: checkAddOperator uses false param for `isPromoting`.
 	// This is used to keep check logic before fixing issue #4946,
 	// but maybe user want to add operator when waiting queue is busy
-	if oc.exceedStoreLimitLocked(ops...) || !oc.checkAddOperator(false, ops...) {
+
```
Comment (HuSharp) [inline pkg/schedule/operator/operator_controller.go:333]:
```suggestion
```

File: pkg/schedule/operator/operator_controller.go [hunk 3]
```diff
@@ -416,26 +436,28 @@ func (oc *Controller) checkAddOperator(isPromoting bool, ops ...*Operator) bool
...
 	}
 	expired := false
+	var reason CancelReasonType
 	for _, op := range ops {
 		if op.CheckExpired() {
 			expired = true
+			reason = Expired
 			operatorWaitCounter.WithLabelValues(op.Desc(), "expired").Inc()
 		}
 	}
```
Comment (HuSharp) [inline pkg/schedule/operator/operator_controller.go:459]: How about delete a var 🤔
```suggestion
	for _, op := range ops {
		if op.CheckExpired() {
			reason = Expired
			operatorWaitCounter.WithLabelValues(op.Desc(), "expired").Inc()
		}
	}
	return reason != Expired, reason
```

Comment (bufferflies) [reply pkg/schedule/operator/operator_controller.go]: fixed

**Review by HuSharp — APPROVED** [review]
LGTM!

**Review by nolouch — APPROVED** [review]
lgtm

**Pull request merged**